apps/src/weblab/CdoBramble.js

```diff
@@ -112,7 +126,7 @@ export default class CdoBramble {
     const hasSyncedVersion = this.lastSyncedVersionId === projectVersion;
 
     // Send any new changes to the server.
-    if (hasSyncedVersion && this.recentChanges.length > 0) {
+    if (hasSyncedVersion) {
```

Comment:
note about this change: `this.recursivelySaveChangesToServer` will immediately invoke the callback if there are no changes, which is the behavior we want. before, if you called this method and `hasSyncedVersion` was true but `this.recentChanges` was empty, the callback was never invoked and you'd get an infinite spinner.